moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryProjectDescription.js

```diff
@@ -25,109 +34,117 @@ import { useMutation } from "@apollo/client";
+    PROJECT_UPDATE_DESCRIPTION
+  );
 
   /**
-   * Resets the project description to original value
+   * Switch to view mode on close
    */
   const handleProjectDescriptionClose = () => {
-    setDescription(originalDescription);
     setEditMode(false);
   };
 
   /**
    * Saves the new project description...
    */
-  const handleProjectDescriptionSave = () => {
-    if (description.trim().length > 0) {
-      updateProjectDescription({
-        variables: {
-          projectId: projectId,
-          description: description,
-        },
-        refetchQueries: [{ query: listViewQuery }, "ProjectSummary"],
-      })
-        .then(() => {
-          setEditMode(false);
-          handleSnackbar(true, "Project description updated", "success");
-        })
-        .catch((error) => {
-          handleSnackbar(
-            true,
-            "Error updating project description",
-            "error",
-            error
-          );
+
+  const handleProjectDescriptionSave = ({ description }) => {
+    updateProjectDescription({
+      variables: {
+        projectId: projectId,
+        description: description,
+      },
+      refetchQueries: [{ query: listViewQuery }],
+    })
+      .then(() => {
+        refetch().then(() => {
           handleProjectDescriptionClose();
+          handleSnackbar(true, "Project description updated", "success");
         });
-      setEditMode(false);
-    } else {
-      setDescription(description.trim());
-    }
-  };
-
-  /**
-   * Updates the description state
-   * @param {Object} e - Event object
-   */
-  const handleProjectDescriptionChange = (e) => {
-    setDescription(e.target.value);
+      })
+      .catch((error) => {
+        handleSnackbar(
+          true,
+          "Error updating project description",
+          "error",
+          error
+        );
+        handleProjectDescriptionClose();
+      });
   };
 
   return (
     <Grid item xs={12} className={classes.fieldGridItem}>
       <Typography className={classes.fieldLabel}>Description</Typography>
+
       <Box
         display="flex"
         justifyContent="flex-start"
         className={classes.fieldBox}
         flexWrap="nowrap"
         alignItems="center"
+        component="form"
+        onSubmit={handleSubmit(handleProjectDescriptionSave)}
       >
-        {editMode && (
+        {editMode ? (
           <>
-            <TextField
+            <ControlledTextInput
               variant="standard"
-              autoFocus
               fullWidth
-              multiline={true}
+              autoFocus
+              multiline
               minRows={4}
-              error={description.length < 1}
-              id="moped-project-description"
-              label={null}
-              onChange={handleProjectDescriptionChange}
-              value={description}
-              helperText={
-                description.length < 1 ? "Description cannot be blank" : ""
-              }
+              maxRows={10}
```

Comment:
I added this maximum so that long strings don't endlessly grow in this input. You can compare this behavior to staging where you can paste super long strings and they endlessly grow the page.